### PR TITLE
Add `OakLink`

### PR DESCRIPTION
**Pull request by carlmw**
Adds the blue flavour of link. With a little refactoring this can be re-used to build the secondary flavour as in the designs. 

Initially needed for the copyright notice in the pupil intro page

Storybook: https://deploy-preview-102--lively-meringue-8ebd43.netlify.app/?path=/docs/components-ui-oaklink--docs
Figma: https://www.figma.com/file/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?type=design&node-id=3473-12290&mode=dev

Screenshot 2024-02-12 at 14 19 32

### CODE REVIEW

File: src/components/integrated/OakLessonVideoTranscript/__snapshots__/OakLessonVideoTranscript.test.tsx.snap [hunk 1]
```diff
@@ -88,13 +88,13 @@ exports[`OakLessonVideoTranscript matches snapshot 1`] = `
...
-  -moz-letter-spacing: -0.005rem;
-  -ms-letter-spacing: -0.005rem;
-  letter-spacing: -0.005rem;
+  font-weight: 600;
+  font-size: 1rem;
+  line-height: 1.25rem;
+  -webkit-letter-spacing: 0.0115rem;
+  -moz-letter-spacing: 0.0115rem;
+  -ms-letter-spacing: 0.0115rem;
+  letter-spacing: 0.0115rem;
```
Comment (carlmw) [inline src/components/integrated/OakLessonVideoTranscript/__snapshots__/OakLessonVideoTranscript.test.tsx.snap:97]: rug was pulled on this in a previous merge so main was broken. This fixes the broken snapshot

File: src/components/ui/OakLink/OakLink.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  font: inherit;
+  background: none;
+  border: none;
+  text-decoration: underline;
+  cursor: pointer;
+
+  color: ${parseColor("text-link-active")};
+  ${StyledOakIcon} {
+    filter: ${parseColorFilter("text-link-active")};
+  }
```
Comment (carlmw) [inline src/components/ui/OakLink/OakLink.tsx:40]: This ensures that the icon matches the colour of the text

File: src/components/utils/polymorphic.ts [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
 };
 
 export type PolymorphicPropsWithoutRef<C extends ElementType> = ElementProp<C> &
   ComponentPropsWithoutRef<C>;
+
+export type PolymorphicRef<C extends React.ElementType> =
+  ComponentPropsWithRef<C>["ref"];
+
+export type PolymorphicPropsWithRef<C extends ElementType> = ElementProp<C> &
+  ComponentPropsWithoutRef<C> & { ref?: PolymorphicRef<C> };
```
Comment (carlmw) [inline src/components/utils/polymorphic.ts:18]: Some new utility types to support forwarding refs through polymorphic components.

File: src/components/ui/OakLoadingSpinner/OakLoadingSpinner.tsx [hunk 4]
```diff
@@ -31,7 +31,6 @@ const StyledLoadingSpinner = styled.span<OakLoadingSpinnerProps>`
   --inner-width: calc(var(--width) / 10 * 8);
   --thickness: calc(var(--width) / 12);
 
-  position: absolute;
```
Comment (carlmw) [inline src/components/ui/OakLoadingSpinner/OakLoadingSpinner.tsx:34]: This was throwing off the position of the spinner in a flex container — it doesn't appear to be needed at all looking at all the examples which appear the same.

**Review by kimon-satan — APPROVED** [review]

**Comment by messmike** [comment]
@carlmw looking good, only thing from me is that the icon doesn't seem to change to match the colour of the text on 'pressed' state: 
Screenshot 2024-02-12 at 2 28 18 pm (2)

**Comment by carlmw** [comment]
@messmike ahh yes. I've taken a look and the problem is in the icon. It contains a fill that isn't 100% black.

Screenshot 2024-02-12 at 15 35 10

In OWA and the component library all icons are coloured with a CSS filter. 

Like this:
https://github.com/oaknational/oak-components/blob/86457d55bae536e823ee3e0cd9d4c117d2e61360/src/styles/theme/color.ts#L68-L72

It's a bit of a hack to achieve a colour match but it relies on the icon being completely black or the colour shifting gets out of whack.

I've updated the chevron-left and chevron-right icons so that they have a 100% black fill. This is something we'll have to look out for in future when exporting icons.

**Pull request merged**